File: src/main/java/com/challenge/interview/calendar/model/AvailabilityEntity.java
package com.challenge.interview.calendar.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;

@AllArgsConstructor
@Data
@NoArgsConstructor
@Entity
public class AvailabilityEntity {

    public AvailabilityEntity(OffsetDateTime from, OffsetDateTime to, String calendarId, String role) {
        this.from = from;
        this.to = to;
        this.calendarId = calendarId;
        this.role = role;
    }

    @Column(name = "fromDate", nullable = false)
    private OffsetDateTime from;

    @Column(name = "toDate", nullable = false)
    private OffsetDateTime to;

    @Column(nullable = false)
    private String calendarId;

    @Column(nullable = false)
    private String role;

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    public TimeInterval toTimeInterval() {
        return new TimeInterval(this.from, this.to);
    }
}

File: src/main/kotlin/com/challenge/interview/calendar/model/AvailabilityEntity.kt
package com.challenge.interview.calendar.model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import java.time.OffsetDateTime

@Entity(name = "availability")
class AvailabilityEntity() {
    public constructor(from: OffsetDateTime,
                       to: OffsetDateTime,
                       calendarId: String,
                       role: String): this() {
        this.from = from
        this.to = to
        this.calendarId = calendarId;
        this.role = role;
    }

    @Column(name = "fromDate", nullable = false)
    lateinit var from: OffsetDateTime;

    @Column(name = "toDate", nullable = false)
    lateinit var to: OffsetDateTime;

    @Column(nullable = false)
    lateinit var calendarId: String;

    @Column(nullable = false)
    lateinit var role: String;

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long? = null;

    fun toTimeInterval(): TimeInterval {
        return TimeInterval(this.from, this.to)
    }
}